### PR TITLE
Engineering Goggles set to T-ray can now see wires when hacking. 

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request

* Engineering goggles now reveal wires when hacking objects without random wires or airlocks
* Engineers (including chief) now spawn with engineering goggles

## Why It's Good For The Game

Major QoL improvement for engineers that softens the effects of #6920 and makes it easier to do their job in the wake of the changes it brought. Also adds a new angle for antagonists to approach greytiding when they want or need to.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

T-ray mode active:
image

Not in T-ray mode:
image

</details>

## Changelog
:cl:
add: Engineering goggles now reveal wires when in T-ray mode.
tweak: Engineers, including the chief engineer, now spawn with engineering goggles equipped instead of in their lockers. Other Loose spawns are unchanged.
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
@FortSumter yo

**Comment by PestoVerde322** [comment]
What about the normal T-ray? Other than that, looking forward to this!

File: code/datums/wires/_wires.dm [hunk 1]
```diff
@@ -252,9 +252,14 @@
...
-	// Station blueprints do that too, but only if the wires are not randomized.
-	else if(user.is_holding_item_of_type(/obj/item/areaeditor/blueprints) && (!randomize || holder_type == /obj/machinery/door/airlock))
-		reveal_wires = TRUE
+	// Station blueprints and T-ray engineer goggles can do that too, but not with everything
+	else if(!randomize || holder_type == /obj/machinery/door/airlock)
+		if(user.is_holding_item_of_type(/obj/item/areaeditor/blueprints))
+			reveal_wires = TRUE
+		else if(istype(user.get_item_by_slot(ITEM_SLOT_EYES), /obj/item/clothing/glasses/meson/engine))
+			var/obj/item/clothing/glasses/meson/engine/goggles = user.get_item_by_slot(ITEM_SLOT_EYES)
+			if(goggles.mode == "t-ray")
```
Comment (PowerfulBacon) [inline code/datums/wires/_wires.dm]: Checking against a string is disgusting

**Comment by bluezorua** [comment]
no need to make engineers spawn with engi goggles since every engineer locker spawns with a pair+ there's also engivend

**Comment by Dejaku51** [comment]
> Chief engineer now spawns with insulated gloves

they have gloves in locker

File: code/datums/wires/_wires.dm (hunk 1 above)
Comment (Dejaku51) [reply code/datums/wires/_wires.dm]: you can use `MODE_TRAY`

**Comment by PowerfulBacon** [comment]
> tweak: Chief Engineer now spawns with insulated gloves instead of black ones

Does he?

**Comment by Rukofamicom** [comment]
> Does he?

If I remembered to push off of remote, yes. Did that now lmao

File: code/datums/wires/_wires.dm (hunk 1 above)
Comment (Rukofamicom) [reply code/datums/wires/_wires.dm]: Not unless I make that define global, which I didn't know how to do until just now.

**Comment by FortSumter** [comment]
If you could also make some interaction with the handheld t ray scanner that would be nice. Maybe make it take 15 seconds for the person to scan the door or something. This way  normal crew can hack doors again but there's a static minimum time required to do it.

**Comment by Rukofamicom** [comment]
> If you could also make some interaction with the handheld t ray scanner that would be nice.

Working out the details of what exactly to do with it still, but it is planned. Was originally going to be part of this PR, but I hit a big enough bump that I decided pushing this as-is was better than waiting until I found a way to apply a similar usage to the handheld one.

File: code/modules/jobs/job_types/chief_engineer.dm [hunk 2]
```diff
@@ -48,8 +48,9 @@
 	ears = /obj/item/radio/headset/heads/ce
 	uniform = /obj/item/clothing/under/rank/engineering/chief_engineer
 	shoes = /obj/item/clothing/shoes/sneakers/brown
+	glasses = /obj/item/clothing/glasses/meson/engine
 	head = /obj/item/clothing/head/hardhat/white
-	gloves = /obj/item/clothing/gloves/color/black
+	gloves = /obj/item/clothing/gloves/color/yellow
```
Comment (Crossedfall) [inline code/modules/jobs/job_types/chief_engineer.dm]:
```suggestion
	gloves = /obj/item/clothing/gloves/color/black
```

Don't mix unrelated changes.

**Review by Crossedfall — DISMISSED** [review]
Keep changes related to the actual scope of the PR, don't add in unrelated stuff.

File: code/modules/jobs/job_types/chief_engineer.dm (hunk 2 above)
Comment (Rukofamicom) [reply code/modules/jobs/job_types/chief_engineer.dm]: It's related to door hacking being easier for engineers, but alright I'll make an entirely separate PR for the CE gloves. 

See the branch name - this was intended to be a general improvement to hacking for engineers.

**Review by PowerfulBacon — DISMISSED** [review]
Looks good to me

**Comment by PowerfulBacon** [comment]
This does however have the interesting side effect of making these glasses even more sought after for tiding than insulated gloves, considering they not only let you freely hack a door but let you see every single wire and what it does, making hacking instant.
Also the fact that these are infinitely printable while insulated gloves aren't could make them a bit too powerful.

**Comment by PowerfulBacon** [comment]
Honestly, I think this is too powerful. I think the T-Ray scanner should have a 5 second cooldown use in which it can be used to check if a wire will shock you, or instead have something that can reveal how the wires operate for a single device after a scanning period and is not easilly accessible.

**Comment by Rukofamicom** [comment]
@PowerfulBacon see post above:
I can make them non printable if that will sway you?

> I'd prefer if there were none available outside engineering at round start, meaning you will either have to be an engineer, bribe an engineer for a pair, break into engineering and bust a locker, or kill an engineer to get a pair before 30 min. This gives antags a reason to antag if they want psudeo AA and not spend their bad boy points on it.

That is currently how it is and I kinda agreed tbh. I don't think the engi lathe is hard to get to. Goggles tell you the wires, but you will still need other tools on top of *not* having sunglasses while you're using them on a door.

**Comment by FortSumter** [comment]
>  I don't think the engi lathe is particularly hard to get to
I would have to look at the station webmaps to be certain, but on most maps I can think of besides Box the lathe is locked behind at least 2 sets of doors with no wall adjacent to a maintenance corridor, which means you will be breaking into engineering from an obvious main hall just to get in the lathe room, and all of the engineering staff pass it on their way out of their departments main door. The lathe is both difficult to reach _and_ highly visible, making it more difficult to get to than even securities lathe on some maps.

If they were kept printable there would be no way in hell you'd get a pair without a massive amount of risk involved unless you are an antag and have something like an emag, in which case I can't understand why you'd want the goggles to begin with.

**Comment by PowerfulBacon** [comment]
I don't know, I just think they are way too powerful and that wouldn't change if they weren't printable. I think there should be a safe way to hack, but it should be time consuming. This is both 100% safe and instant.

**Comment by FortSumter** [comment]
Bacon. This is literally what the engineering skillships were meant to do, and something that is absolutely necessary for engineers to have now that all door wiring is random. If you didn't want people to go down this road you should have considered how making all the doors wires random would effect game balance.

The silicon ability to bolt closed doors is now one of the most powerful abilities in the game. It can catch you between two sets of doors and even if you're an engineer and have the tools to hack your way out, you might as well deconstruct the door outright as that's easier than figuring out what the bolt wire is.

**Comment by FortSumter** [comment]
It makes no sense to have destroying a door be more efficient than forcing it open.

**Comment by PowerfulBacon** [comment]
I don't actually have a problem with the door balance right now. I think it is better balanced as fully randomised, since it means a consistent hacking time for all doors rather than instant hacks.
The only balance changes that I would want is a slow but safe way to hack open doors, for example using TRay scanner to activate the door safely but with a 5 second cooldown.
Currently engineers can still reasonably deconstruct doors, they just have to pulse until the door breaks and then crowbar it without power almost the same as before, except they have to find the power wire first.

**Comment by FortSumter** [comment]
As a player I still viciously hate the random door wires. It's just too easy for silicons to lock you out/ in rooms and makes it mandatory for antags to buy a method of bypassing the door. If you're a BB or rev you are f u c k e d.
I understand it had to be done for game balance, but if we don't have some kind of method, you've just tipped the scales the other way AND made doors a pain in the ass for engineers.

**Comment by itsmeow** [comment]
This is a good change for balance, it lets engineers actually do their jobs and offers antags a way to hack doors while still not easily obtainable roundstart (you can't get the goggles in public tool storage like you can insuls, tools, etc.)

Yes, as an engineer, I find it a lot easier to deconstruct a door with an RCD than try to fix the mess that is its wires if somebody tampered with it. Or I'd just call a borg to fix it, because they can actually interface with doors in a sensible manner.

File: code/__DEFINES/hud.dm [hunk 3]
```diff
@@ -9,3 +9,12 @@
 #define HUD_STYLE_NOHUD 3 //! No hud (for screenshots)
 
 #define HUD_VERSIONS 3	//! Used in show_hud(); Please ensure this is the same as the maximum index.
+
+// Modes used by /obj/item/clothing/glasses/meson/engine
+
+
+#define MODE_NONE ""
+#define MODE_MESON "meson"
```
Comment (itsmeow) [inline code/__DEFINES/hud.dm]: This does not belong in the HUD defines.

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Comment by itsmeow** [comment]
My thoughts on the balance of this PR:
It lets engineers actually do their job without begging a silicon to fix the door (good)
It adequately controls the ability for greytiders to obtain said item, while still allowing antags the opportunity (good)
Anyone who desires said item would probably be better off getting insuls and then spamming wirecut until they bolt it, then use a multitool unbolt, cut everything else. Getting the goggles is not easy.

The door randomization exists to prevent tiding being so easy - which it achieved. But it failed to consider that doors do need to be fixed and the wires need to be usable.

Door wires have unique functions, engineers NEED to have a way to find out which is which - just having a T-ray open a door is not enough.

I think the effects of this change on non-engineering is very minor. Perhaps it could stay randomized on command or high security doors?

File: code/__DEFINES/hud.dm (hunk 3 above)
Comment (Rukofamicom) [reply code/__DEFINES/hud.dm]: why... not? 

I was indirectly asked to make these strings global defines so that I wasn't checking against a string by Bacon in an earlier review, and they have to be defined before the check occurs as I learned. This is where all other global hud definitions are defined?

image

Comment (itsmeow) [reply code/__DEFINES/hud.dm]: yes, you've put them in a bad spot, there are tons of defines files, there's got to be one that fits these better. This file seems to be for onscreen HUDs, not the worn kind.

Comment (Rukofamicom) [reply code/__DEFINES/hud.dm]: Would you prefer it be in a completely unique file? I can't keep them where they originally were because even if I make them global that file compiles *after* _wires.dm which results in this snippet of code failing to compile due to `MODE_TRAY` being undefined:

```
if(goggles.mode == MODE_TRAY)
	reveal_wires = TRUE
```

Comment (itsmeow) [reply code/__DEFINES/hud.dm]: Just looked for you,
sight.dm would be a good spot. Borg mesons are there.
https://github.com/BeeStation/BeeStation-Hornet/blob/master/code/__DEFINES/sight.dm

Comment (Rukofamicom) [reply code/__DEFINES/hud.dm]: :+1:

Comment (Rukofamicom) [reply code/__DEFINES/hud.dm]: Done

**Review by itsmeow — APPROVED** [review]

**Comment by hotimep** [comment]
(probably) op as hell, the only other two items in the game that let you see wires are CE's blueprints and the abductors multitool, iffy as it being available roundstart to every engineer and random guy that breaks into engineering, but it would be cool as a thing that can be researched

**Comment by FortSumter** [comment]
It's not a case if its OP or not, but if its necessary and it is thanks to random door wires. The engineers should have a reliable method to hack doors as it's part of their job, and this will make it easier for noob engis while also giving antags a tool to make hacking feasible. As for the tools that have this function already, the blueprints are CE only and has other functions and the abductor tool is necessary for their antag. Personally I wouldn't of approved the fucking stupid random door wires to begin with, but here we are.

**Comment by PowerfulBacon** [comment]
It's not necessary, random door wires do not make it so you require tools to see all the wires. Random door wires make the time to hack on all doors relatively high, however if you have insulated gloves it is still trivial to hack a door open, or fix a door.
Looking at it, random wires were a pretty good choice for balance considering that hacking now becomes a time consuming process rather than free access to all places on the station after you test 1 random door.
The only thing I think that should be changed, is that I think hacking should be made safer, IE hacking is a time consuming process but without gloves you won't get shocked as often as you do now.

I'm putting a close key on this, since I think making an item that can instantly determine all wires in a door when worn with no cooldown defeats the purpose of having random wire doors. I think something like having the T-Ray scanner have the ability to test what a wire does without pulsing it on a cooldown would be a better solution, since it would allow engineers to determine which wire they need to pulse and allow anyone with the tool a guaranteed but slow way to get through non-reinforced doors.

**Comment by FortSumter** [comment]
Another option I've been made aware of from other servers if you still insist on keeping this fucking stupid random door wiring (as it is indeed both unrealistic and fucking stupid) would be to make it so the doors don't shock at all when hacking or when breaking the door down and give textual indications of what pulsing the wire did like it does when you hack MULEs. This would make hacking a door more like a minigame than the most fucking annoying thing to ever exist. Itd also give people with hulk, simple mobs, and anyone with a sufficiently powerful weapon the ability to just destroy the door without harming themself, as all the does is limit fun.

**Comment by Rukofamicom** [comment]
Branch will stay up if anyone else wants to modify, but one key is enough for me to take it off the to-do list.
